File: TaskManager/TaskManagerSolution/src/ChangeIssueStatusCommand.java
public class ChangeIssueStatusCommand extends Command{
    private final IssueRepository repository;
    private final String issueId;
    private final Status newStatus;
    private Status oldStatus;

    public ChangeIssueStatusCommand(IssueRepository repository, String issueId, Status newStatus) {
        super("Changed issue status to: " + newStatus, CommandType.UPDATE);
        this.repository = repository;
        this.issueId = issueId;
        this.newStatus = newStatus;
    }

    @Override
    public void execute() {
        Issue issue = repository.getIssue(issueId);
        oldStatus = issue.getStatus();
        issue.setStatus(newStatus);
    }

    @Override
    public void undo() {
        if (oldStatus != null) {
            Issue issue = repository.getIssue(issueId);
            issue.setStatus(oldStatus);
        }
    }

}

File: TaskManager/MyTaskManager/ChangeIssueStatusCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTaskManager
{
    public class ChangeIssueStatusCommand : Command
    {
        public IssueRepository repository;
        public int issueId;
        public Status newStatus;
        public Status oldStatus;

        public ChangeIssueStatusCommand(IssueRepository repository, int issueId, Status newStatus)
        {
            this.repository = repository;
            this.issueId = issueId;
            this.newStatus = newStatus;
            this.Type = CommandType.Update;
            this.Description = $"Changed issue id{issueId} status to {newStatus}";
        }

        public override void Execute()
        {
            Issue issue = repository.GetIssue(issueId);
            if (issue != null)
            {
                oldStatus = issue.Status;
                issue.Status = newStatus;
                issue.LastModifiedDate = DateTime.Now;
                repository.UpdateIssue(issue);
            }
        }

        public override void Undo()
        {
            Issue issue = repository.GetIssue(issueId);
            if (issue != null)
            {
                issue.Status = oldStatus;
                issue.LastModifiedDate = DateTime.Now;
                repository.UpdateIssue(issue);
            }
        }
    }
}
